ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Initiator.java

```diff
@@ -259,15 +268,32 @@ private void scheduleCompactionIfRequired(CompactionInfo ci, Table t, Partition
     }
   }
 
-  private Table resolveTableAndCache(CompactionInfo ci) throws MetaException {
-    if (tableCache.isPresent()) {
+  private String getPoolName(CompactionInfo ci, Table t) throws Exception {
+    String poolName = t.getParameters().get(Constants.HIVE_COMPACTOR_WORKER_POOL);
+    if (StringUtils.isBlank(poolName)) {
+      poolName = obtainObject(ci.dbname, () -> resolveDatabase(ci)).getParameters().get(Constants.HIVE_COMPACTOR_WORKER_POOL);
```

Comment:
same here, is it possible that we might call `getParameters()` on null object

Reply:
same as above